mk/config.mk

```diff
@@ -49,7 +41,7 @@ include mk/$(PLATFORM).mk
 
 CFLAGS += \
 	-O3 \
-	-std=gnu99 \
+	-std=gnu17 \
```

Comment:
Why is this change needed? We've been using C99 for a long time in many projects and changing this here may lead to compatibility issues down the road. I guess it's needed for some of the dependencies? Can we maybe compile those dependencies separately with different CFLAGS, so that our code is still compiled using C99?

Reply:
esp-idf need to be compiled with C17, but you're right, since it's only used for hal, we should just compile the hal static library with C17 instead.